spec/index.bs

```diff
@@ -4183,41 +4189,119 @@ has a default layout created and used instead.
 
 ### <dfn dictionary>GPUProgrammableStage</dfn> ### {#GPUProgrammableStage}
 
+A {{GPUProgrammableStage}} describes the entry point in the user-provided
+{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+
 <script type=idl>
 dictionary GPUProgrammableStage {
     required GPUShaderModule module;
     required USVString entryPoint;
+    record<USVString, GPUPipelineConstantValue> constants;
 };
+
+typedef double GPUPipelineConstantValue; // May represent WGSL's bool, f32, i32, u32.
 </script>
 
-A {{GPUProgrammableStage}} describes the entry point in the user-provided
-{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+<dl dfn-for=GPUProgrammableStage dfn-type=dict-member>
+    : <dfn>constants</dfn>
+    ::
+        Specifies the values of [=pipeline-overridable=] constants which are [=statically accessed=]
```

Comment:
What happens if it's not statically accessed? I.e. I'm toying with the shader, commenting out bits here and there. Would the pipeline suddenly become invalid just because my host API code passes an extra constant?

Reply:
This is a good point. I guess we should allow any pipeline-overridable constant in the entire shader module? The name is already unique to the entire module so that would work.

Reply:
right, yes